### PR TITLE
address double install with both hpi and jpi file being present

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/jenkins/issues/101

@bparees PTAL

### CODE REVIEW

**Comment by bparees** [comment]
looks like we need to change the permissions on those files so they can be overwritten when the plugins.txt processing goes to install them again:

```
I0610 17:13:28.843292 04109 sti.go:603] ---> Copying repository files ...
I0610 17:13:28.853050 04109 sti.go:603] ---> Installing Jenkins 3 plugins using /opt/openshift/plugins.txt ...
I0610 17:13:28.991341 04109 sti.go:603] Downloading credentials-1.23 ...
I0610 17:13:29.445514 04109 sti.go:603] Failed to download credentials-1.23:
I0610 17:13:29.446093 04109 sti.go:603] curl: (23) Failed writing body (0 != 2664)
I0610 17:13:29.447064 04109 sti.go:603] Failed to install plugins.
```

**Comment by gabemontero** [comment]
added a fix-permission call for `/opt/openshift/plugins`

**Comment by gabemontero** [comment]
Same error even with the fix-permission call ...

**Comment by gabemontero** [comment]
trying an explicit chmod on the jpi files

File: 1/Dockerfile.rhel7 [hunk 1]
```diff
@@ -47,14 +47,16 @@ ADD ./contrib/s2i /usr/libexec/s2i
...
     # version.
-    touch /opt/openshift/plugins/credentials.hpi.pinned && \
-    ln -s /usr/lib64/jenkins/durable-task.hpi /opt/openshift/plugins/durable-task.hpi && \
-    ln -s /usr/lib64/jenkins/kubernetes.hpi /opt/openshift/plugins/kubernetes.hpi && \
+    touch /opt/openshift/plugins/credentials.jpi.pinned && \
+    ln -s /usr/lib64/jenkins/durable-task.hpi /opt/openshift/plugins/durable-task.jpi && \
+    ln -s /usr/lib64/jenkins/kubernetes.hpi /opt/openshift/plugins/kubernetes.jpi && \
     /usr/local/bin/fix-permissions /opt/openshift && \
+    /usr/local/bin/fix-permissions /opt/openshift/plugins && \
+    chmod -R g+rw /opt/openshift/plugins/*.jpi && \
```
Comment (bparees) [inline 1/Dockerfile.rhel7]: that this appears to have worked means something is going wrong w/ the fix-permissions script..

in fact given that we're making the call on /opt/openshift, adding another call on /opt/openshift/plugins also shouldn't be necessary.

I wonder if chmod/chgrp -R doesn't follow symlinks or something.

that appears to be the case:
```
zahir:/tmp$ ln -s /tmp/b a/blink
zahir:/tmp$ ls -l a
total 0
lrwxrwxrwx. 1 bmp bmp 6 Jun 12 23:24 blink -> /tmp/b

zahir:/tmp$ ls -l b
-rw-rw-r--. 1 bmp bmp 0 Jun 12 23:24 b

zahir:/tmp$ chmod -R a+rwx /tmp/a
zahir:/tmp$ ls -l b
-rw-rw-r--. 1 bmp bmp 0 Jun 12 23:24 b

zahir:/tmp$ cd /tmp/a
zahir:/tmp/a$ chmod a+rwx *
zahir:/tmp/a$ ls -l /tmp/b
-rwxrwxrwx. 1 bmp bmp 0 Jun 12 23:24 /tmp/b
```

http://ubuntuforums.org/showthread.php?t=2046341

we may need to do this entirely via a "find" operation.

Comment (gabemontero) [inline 1/Dockerfile.rhel7]: yep, good catch on the link thing ... and yep, I got a find / exec chmod to work using your sample above.

File: 1/contrib/jenkins/fix-permissions [hunk 2]
```diff
@@ -2,5 +2,6 @@
 # Fix permissions on the given directory to allow group read/write of 
 # regular files and execute of directories.
 chgrp -R 0 $1
-chmod -R g+rw $1
+#chmod -R g+rw $1
+find $1 -exec chmod g+rw {} \;
```
Comment (bparees) [inline 1/contrib/jenkins/fix-permissions]: guessing chgrp will need the same treatment?

Comment (gabemontero) [inline 1/contrib/jenkins/fix-permissions]: yeah probably ... i'll wait until the upcoming test completes successfully for the find / chmod before officially pulling that trigger

**Comment by gabemontero** [comment]
OK, all use of `-R` replaced with `find -exec`

**Comment by bparees** [comment]
looks good to me, can you follow up w/ this same change to the other copies o fix-permissions?  (one in s2i-base, and possibly one in each of the DB images?)

**Comment by gabemontero** [comment]
On Mon, Jun 13, 2016 at 9:22 AM, Ben Parees <notifications@github.com>
wrote:

> looks good to me, can you follow up w/ this same change to the other
> copies o fix-permissions? (one in s2i-base, and possibly one in each of the
> DB images?)
>
sure I'll take a look

**Pull request merged**